src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/CannotAddOrRemoveStaticKeyword.cs

```diff
@@ -0,0 +1,92 @@
+    {
+        private readonly IRuleSettings _settings;
+
+        public CannotAddOrRemoveStaticKeyword(IRuleSettings settings, IRuleRegistrationContext context)
+        {
+            _settings = settings;
+            context.RegisterOnMemberSymbolAction(RunOnMemberSymbol);
+            context.RegisterOnTypeSymbolAction(RunOnTypeSymbol);
+        }
+
+        private void RunOnMemberSymbol(ISymbol? left, ISymbol? right, MetadataInformation leftMetadata, MetadataInformation rightMetadata, IList<CompatDifference> differences)
+        {
+            // Members must exist
+            if (left is null || right is null)
+            {
+                return;
+            }
+
+            // Check if static modifier changed on the member
+            if (left.IsStatic && !right.IsStatic)
+            {
+                // Removing static is always breaking (both binary and source breaking)
+                differences.Add(new CompatDifference(
+                    leftMetadata,
+                    rightMetadata,
+                    DiagnosticIds.CannotRemoveStaticFromMember,
+                    string.Format(Resources.CannotRemoveStaticFromMember, right),
+                    DifferenceType.Removed,
+                    right));
+            }
+            else if (!left.IsStatic && right.IsStatic)
+            {
+                // Adding static is always breaking (binary breaking)
+                differences.Add(new CompatDifference(
+                    leftMetadata,
+                    rightMetadata,
+                    DiagnosticIds.CannotAddStaticToMember,
+                    string.Format(Resources.CannotAddStaticToMember, right),
+                    DifferenceType.Added,
+                    right));
+            }
+        }
+
+        private void RunOnTypeSymbol(ITypeSymbol? left, ITypeSymbol? right, MetadataInformation leftMetadata, MetadataInformation rightMetadata, IList<CompatDifference> differences)
+        {
+            // Types must exist
+            if (left is null || right is null)
+            {
+                return;
+            }
+
+            // Check if static modifier was added to the type
+            // Adding static to a type is breaking since static types cannot be used as 
+            // return types, generic parameters, or arguments
+            if (!left.IsStatic && right.IsStatic)
+            {
+                differences.Add(new CompatDifference(
+                    leftMetadata,
+                    rightMetadata,
+                    DiagnosticIds.CannotAddStaticToType,
+                    string.Format(Resources.CannotAddStaticToType, right),
+                    DifferenceType.Added,
+                    right));
+            }
+            // In strict mode, report when static is removed from a type
+            // Removing static from a type is compatible (members will be checked separately)
+            // but developers may want to be aware of this change
```

Comment:
We shouldn't just say "may" - the purpose of strict mode is to make any changes breaking (EG: diffing reference to src).  Please update the comment.

```suggestion
            // In strict mode, report when static is removed from a type
```